### ISSUE TITLE
Conflicting Version of Python Dependency in build.txt 

### ISSUE THREAD

**Issue by ozer550** [learningequality/kolibri#10502]
## Observed behavior
Conflicting version of the library [setuptools](https://github.com/learningequality/kolibri/blob/e332e6937b43f42b05c4b462309b48175865df64/requirements/build.txt#L7) in [build.txt](https://github.com/learningequality/kolibri/blob/e332e6937b43f42b05c4b462309b48175865df64/requirements/build.txt) file causing error while installing requirements given in [test.txt](https://github.com/learningequality/kolibri/blob/e332e6937b43f42b05c4b462309b48175865df64/requirements/test.txt) and [docs.txt](https://github.com/learningequality/kolibri/blob/e332e6937b43f42b05c4b462309b48175865df64/requirements/docs.txt).   

## Errors and logs
Run in order as mentioned in the [guide](https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#install-python-dependencies)
```
pip install -r requirements/build.txt --upgrade
pip install -r requirements/test.txt --upgrade
Log:

Requirement already satisfied: beautifulsoup4==4.8.2 in /home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages (from -r requirements/test.txt (line 2)) (4.8.2)
Collecting factory-boy==2.7.0
  Using cached factory_boy-2.7.0-py2.py3-none-any.whl (38 kB)
Collecting fake-factory==0.5.7
  Using cached fake_factory-0.5.7-py2.py3-none-any.whl (484 kB)
Collecting coverage==4.5.1
  Using cached coverage-4.5.1.tar.gz (379 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 14, in <module>
        File "/home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages/setuptools/__init__.py", line 20, in <module>
          from setuptools.dist import Distribution, Feature
        File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
        File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
        File "/home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages/_virtualenv.py", line 89, in exec_module
          old(module)
        File "/home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages/setuptools/dist.py", line 35, in <module>
          from setuptools.depends import Require
        File "/home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages/setuptools/depends.py", line 7, in <module>
          from .py33compat import Bytecode
        File "/home/ozer/.virtualenvs/kolibri/lib/python3.9/site-packages/setuptools/py33compat.py", line 55, in <module>
          unescape = getattr(html, 'unescape', html_parser.HTMLParser().unescape)
      AttributeError: 'HTMLParser' object has no attribute 'unescape'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
## Expected behavior
All the dependencies in the requirements file should get installed without conflicts in any order.

## Steps to reproduce
1. Create a virtual environment to install python dependencies.
2. Run `pip install -r requirements/build.txt --upgrade`
3. Then run `pip install -r requirements/test.txt --upgrade`

## Context
OS: Ubuntu 22.04.2 LTS
Kolibri version: Latest
…

**Comment by rtibbles** [comment]
@ozer550 - if you run the different installation commands without the `--upgrade` flag, does the issue still occur?

**Comment by bjester** [comment]
@rtibbles yeah I think @ozer550 can confirm, but I also ran this without `--upgrade` in a fresh environment I used for setup testing and it still occurred. Seems the setuptools version in build.txt isn't compatible with a dependency in test.txt/docs.txt.

**Comment by rtibbles** [comment]
The simplest fix here might be to just remove the recommendation to install the docs requirements then, although fixing the incompatibility would be nice for local docs building tests.

**Comment by zrg228** [comment]
@rtibbles would the fix for the actual code mentioned in the #10489  just closed not help here as well?

**Comment by rtibbles** [comment]
Monkey patching external libraries should be a last resort, and as this is happening during the installation process, there's not a huge window of opportunity to do so. Updating dependencies so that they do not collide with each other in this way is far more preferable.

**Comment by ozer550** [comment]
@rtibbles yes sir, I have tested this without the `--upgrade ` flag and it still didn't work for me. As mentioned by @bjester I can also confirm that the problem is specifically with setuptools library and everything worked smoothly after I upgraded its version.

**Comment by rtibbles** [comment]
OK, unfortunately, I think we have pinned the setuptools version very specifically for the version of PEX that we are using, and for Python 2.7 compatibility - so I think the best fix here is to remove the `build.txt` installation suggestion from the documentation.

**Comment by zrg228** [comment]
> OK, unfortunately, I think we have pinned the setuptools version very specifically for the version of PEX that we are using, and for Python 2.7 compatibility - so I think the best fix here is to remove the `build.txt` installation suggestion from the documentation.

Was there a formal issue set up for this? happy to submit a PR to fix the docs.

**Comment by zrg228** [comment]
> @rtibbles yes sir, I have tested this without the `--upgrade ` flag and it still didn't work for me. As mentioned by @bjester I can also confirm that the problem is specifically with setuptools library and everything worked smoothly after I upgraded its version.

@ozer550 Were you on windows? If so were you using the pyenv-virtualenvironment or was it something else?

**Comment by ozer550** [comment]
> > @rtibbles yes sir, I have tested this without the `--upgrade ` flag and it still didn't work for me. As mentioned by @bjester I can also confirm that the problem is specifically with setuptools library and everything worked smoothly after I upgraded its version.
> 
> @ozer550 Were you on windows? If so were you using the pyenv-virtualenvironment or was it something else?

Hi @zrg228! I was using Ubuntu 22.04.2 LTS for this! And was using pyenv and virtualenvwrapper for creating virtual environments.

**Comment by rtibbles** [comment]
Yeah, I think just generally removing the `build.txt` suggestion from the developer docs completely is best, it's not needed for development, only for the build pipeline.

**Comment by rtibbles** [comment]
Fixed in #10609
